jujugui/static/gui/src/app/store/env/api.js

```diff
@@ -2502,6 +2502,59 @@ YUI.add('juju-env-api', function(Y) {
     },
 
     /**
+      Return the authorized SSH keys for the specified user.
+
+      @method listKeys
+      @param {String} user The user for whom the keys must be listed.
+      @param {Boolean} requestFullKeys Whether to pass full keys to the
+        callback, or just the SSH key fingerprints.
+      @param {Function} callback A callable that must be called once the
+        operation is performed. It will receive two arguments: an error string
+        and the list of keys as strings. If the operation succeeds, the error
+        is null.
+    */
+    listKeys: function(user, requestFullKeys, callback) {
+      if (!user) {
+        callback('no user provided', []);
+        return;
+      }
+      const handler = data => {
+        if (!callback) {
+          console.log('data returned by KeyManager.ListKeys API call:', data);
+          return;
+        }
+        if (data.error) {
+          callback(data.error, []);
+          return;
+        }
+        const results = data.response && data.response.results;
+        if (!results || results.length !== 1) {
```

Comment:
The comment is already there: that should never happen, but we are being defensive, as this is data provided by the network.